.github/workflows/publish_integration_docker_image.yaml

```diff
@@ -0,0 +1,84 @@
+name: Publish integration image to DockerHub
+# Builds and publishes graph integration project to DockerHub
+
+# EXAMPLE USAGE: the following is a yaml job statement using this re-usable workflow
+#   publish-image:
+#    uses: jupiterone/.github/.github/workflows/publish_integration_docker_image.yaml@main
+#    with:
+#      runs_on: ubuntu-latest
+#      package-name: 'jupiterone/graph-kubernetes'
+#      # https://medium.com/@sohail.ra5/github-actions-passing-boolean-input-variables-to-reusable-workflow-call-42d39bf7342e
+#      push-to-registry: ${{ github.event_name != 'pull_request' }}
+#    secrets:
+#      # We HIGHLY recommend that these values are derived using your repos' action secrets
+#      DOCKER_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+#      DOCKER_PASSWORD: ${{ secrets.DOCKERHUB_TOKEN }}
+
+
+on:
+  workflow_call:
+    inputs:
+      runs_on:
+        description: "What github runner should be used (e.g. ubuntu-latest)"
+        required: true
+        type: string
+      package-name:
+        description: "Name of package to publish to DockerHub"
+        required: true
+        type: string
+      push-to-registry:
+        description: "Determines if the built image should be pushed to the registry (default is `false`)"
```

Comment:
Should this default to `true` and be optional?

Reply:
currently it is optional, just defaulting to `false`. I think it is better to not publish unless the user of this workflow explicitly wants to. E.G
`push-to-registry: ${{ github.event_name != 'pull_request' }`
Note that there is some [not so wonderful boolean input-passing issue with workflows](https://medium.com/@sohail.ra5/github-actions-passing-boolean-input-variables-to-reusable-workflow-call-42d39bf7342e)
In light of this, I'd like to encourage `push-to-registry` values that are boolean statements rather than raw values.

Reply:
Interesting. Thanks for digging in.

Is it possible to bake the fact that we only want to run this action when merging to `main` into the action? Additionally, I wonder if we should somehow combine with this with `auto` tagging? What do you think?

Reply:
I can see if I can make ${{ github.event_name != 'pull_request' } the default. I'm not quite sure what you mean by combining this with auto tagging. Can you elaborate?